test/sly-package-fu-tests.el

```diff
@@ -0,0 +1,54 @@
+;; -*- lexical-binding: t; -*-
+(require 'sly-package-fu "contrib/sly-package-fu")
+(require 'sly-tests "lib/sly-tests")
+
+(def-sly-test sly-package-fu-sly-import (initial-defpackage final-defpackage symbol)
+              "Check if importing `import` on `initial-defpackage` results in `final-defpackage."
+              '(("(defpackage :foo
+  (:use :cl))
+(in-package :foo)"
+                 "(defpackage :foo
+  (:use :cl)
+  (:import-from #:cl
+                #:find))
+(in-package :foo)"
+                 "cl:find")
+                ("(defpackage :foo
+  (:use :cl)
+  (:import-from #:cl
+                #:find))
+(in-package :foo)"
+                 "(defpackage :foo
+  (:use :cl)
+  (:import-from #:cl
+                #:position
+                #:find))
+(in-package :foo)"
+                 "cl:position")
+
+                ;; TODO: the output here is incorrect, but we're
+                ;; documenting the current behaviour in this test.
+                ("(defpackage :foo
+  (:use :cl)
+  (:import-from #:bknr.datastore
+                #:find))
+(in-package :foo)"
+                 "(defpackage :foo
+  (:use :cl)
+  (:import-from #:bknr.datastore
+                #:find)
+  (:import-from #:bknr.datastore
+                #:position))
+(in-package :foo)"
+                 "bknr.datastore:position"))
+  (let ((file (make-temp-file "sly-package-fu--fixture")))
+    (with-temp-buffer
+      (find-file file)
+      (lisp-mode)
+      (setq indent-tabs-mode nil)
+      (insert initial-defpackage)
+      (sly-package-fu--add-or-update-import-from-form symbol)
```

Comment:
Relying on an internal detail of `sly-package-fu.el`.  It's not ideal, but OK if nothing else is better.  Wonder if you can't just call the command the user calls with `ert-simulate-command-or-something`.

Reply:
I think in this case, as a unit test, this is the most appropriate thing to do. Usually, it's generally considered okay to use internal details in unit tests, no? In this case, I only want to test the code that modifies the defpackage form, without testing anything else.

Reply:
> sually, it's generally considered okay to use internal details in unit tests, no? 

No, of course not.  Because you're relying on implementation details.  These aren't "unit tests" (i don't know what those are anymore) these are just tests.

What does the user do in real life when using `sly-package-fu`?  You don't have a user's fingers to simulate, but perhaps you can get close enough.

Reply:
Ah, I see your point of view, but, I promise you it'll keep code a lot more testable if we test this directly. We will be able to add a lot of complex tests that are far less brittle. Basically, we don't have to have to be pedantic about "unit" tests, but this code is only testing the logic that updates the defpackage forms, it tests nothing else. This lets us do more complex tests like using weird package names (like `bknr.datastore-dummy`) without having to bring the CL image in sync.

Reply:
Also, despite this being an internal function, I call this function from my `.emacs` because it's super useful in its own right.